extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/VertxHttpProcessor.java

```diff
@@ -185,4 +188,25 @@ void openSocket(ApplicationStartBuildItem start,
     RuntimeInitializedClassBuildItem configureNativeCompilation() {
         return new RuntimeInitializedClassBuildItem("io.vertx.ext.web.handler.sockjs.impl.XhrTransport");
     }
+
+    /**
+     * Register the {@link ExchangeAttributeBuilder} services for native image consumption
+     * 
+     * @param exchangeAttributeBuilderService
+     * @throws BuildException
+     */
+    @BuildStep(onlyIf = NativeBuild.class)
```

Comment:
This should actually happen automatically, as ServiceProviderBuildItem should only be consumed in a native build so it will only run if doing a native build.

Reply:
That makes sense. I'll take care of this while updating this PR.

Reply:
Updated